### ISSUE TITLE
论文中公式（1）中loss的代码实现

### ISSUE THREAD

**Issue by ywjawmw** [cgraywang/deepex#8]
请问论文中公式（1）中loss的实现具体在哪个文件中?仅在Reranking函数中计算了sentence embedding和triple embedding的一阶范数。

**Comment by Magolor** [comment]
Yes you are right. We tested multiple loss functions during experiments. The released code used l1 norm between the sentence embedding and triple embedding indeed. It is not hard to modify it to the NCE-like loss form. We could supplement it in case it is required.

**Comment by ywjawmw** [comment]
Thanks for your apply. I would appreciate it if you could add NCE loss and the relevant training process.

**Comment by Magolor** [comment]
> Thanks for your apply. I would appreciate it if you could add NCE loss and the relevant training process.

Hi! After checking it again, it is worth noting that the code in the Ranking function is used for inference only. It actually does not matter too much what exact function is used for computing inference score as long as the order is maintained.

Our training code uses the loss described in Equation (1). We are still organizing the training scripts and they will be released soon.

**Comment by ywjawmw** [comment]
> Thanks for your apply. I look forward to your release of the training scripts.

**Comment by Magolor** [comment]
The training scripts will be released soon. I'll close this issue and we will pin you once the scripts are released.

**Comment by BeerTai** [comment]
Thank you for great work. Where can I find the training scripts?
